src/containers/LoginForm.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import * as React from 'react';
+import { inject, observer } from 'mobx-react/native';
+import { action, observable } from 'mobx';
+import LoginForm from '../components/LoginForm';
+import { login } from '../authApi';
+
+// todo use mobx-react-form instead?
+@inject('authStore')
+@inject('errorStore')
+@observer
+class LoginFormContainer extends React.Component {
+  @observable userName = '';
+  @observable password = '';
+  @observable loginError = '';
+
+  @action
+  changeUserName = (value: string) => {
+    this.userName = value;
+  };
+
+  @action
+  changePassword = (value: string) => {
+    this.password = value;
+  };
+
+  @action
+  setLoginError = (error: string) => {
+    this.loginError = error;
+  };
+
+  onSubmit = async () => {
```

Comment:
Containers should only fetch data and pass them down as props to the pure component. Business logic should go in action files

Reply:
See above